### ISSUE TITLE
The Dart Analyzer has terminated.

### ISSUE THREAD

**Issue by mezoni** [Dart-Code/Dart-Code#2028]
```
!! PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING !!

Dart Code extension: 3.5.0
VS Code: 1.38.1
Platform: linux
HTTP_PROXY: undefined
NO_PROXY: undefined

[8:40:55 PM] [General] [Info]     Returning SDK path /home/andrew/tools/dart-sdk for dart
[8:41:34 PM] [Analyzer] [Error] Captured exception
[8:41:34 PM] [Analyzer] [Error] NoSuchMethodError: The getter 'iterator' was called on null.
Receiver: null
Tried calling: iterator
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _protocolAvailableSuggestionSet (package:analysis_server/src/domains/completion/available_suggestions.dart:233:35)
#2      createCompletionAvailableSuggestionsNotification.<anonymous closure> (package:analysis_server/src/domains/completion/available_suggestions.dart:78:14)
#3      MappedListIterable.elementAt (dart:_internal/iterable.dart:415:29)
#4      ListIterable.toList (dart:_internal/iterable.dart:219:19)
#5      createCompletionAvailableSuggestionsNotification (package:analysis_server/src/domains/completion/available_suggestions.dart:79:8)
#6      CompletionDomainHandler.setSubscriptions.<anonymous closure> (package:analysis_server/src/domain_completion.dart:497:11)
#7      new DeclarationsTrackerData.<anonymous closure> (package:analysis_server/src/domains/completion/available_suggestions.dart:306:9)
#8      _rootRunUnary (dart:async/zone.dart:1132:38)
#9      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#10     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#11     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#12     _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#13     _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#14     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#15     _rootRun (dart:async/zone.dart:1120:38)
#16     _CustomZone.run (dart:async/zone.dart:1021:19)
#17     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#18     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#19     _rootRun (dart:async/zone.dart:1124:13)
#20     _CustomZone.run (dart:async/zone.dart:1021:19)
#21     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#22     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#23    …
[8:41:34 PM] [Analyzer] [Error] Captured exception
[8:41:34 PM] [Analyzer] [Error] NoSuchMethodError: The getter 'iterator' was called on null.
Receiver: null
Tried calling: iterator
#0      Driver._captureExceptions.errorFunction (package:analysis_server/src/server/driver.dart:677:7)
#1      _CustomZone.handleUncaughtError (dart:async/zone.dart:1003:19)
#2      _CustomZone.runUnaryGuarded (dart:async/zone.dart:933:7)
#3      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#4      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#5      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#6      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#7      _rootRun (dart:async/zone.dart:1120:38)
#8      _CustomZone.run (dart:async/zone.dart:1021:19)
#9      _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#10     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#11     _rootRun (dart:async/zone.dart:1124:13)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#17     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:391:30)
#18     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
#19     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
[8:41:34 PM] [Analyzer] [Error] Captured exception
[8:41:34 PM] [Analyzer] [Error] NoSuchMethodError: The getter 'iterator' was called on null.
Receiver: null
Tried calling: iterator
#0      Driver._captureExceptions.errorFunction (package:analysis_server/src/server/driver.dart:677:7)
#1      _CustomZone.handleUncaughtError (dart:async/zone.dart:1003:19)
#2      _CustomZone.runGuarded (dart:async/zone.dart:925:7)
#3      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#4      _rootRun (dart:async/zone.dart:1124:13)
#5      _CustomZone.run (dart:async/zone.dart:1021:19)
#6      _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#7      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#8      _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#9      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#10     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:391:30)
#11     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
#12     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
[8:41:34 PM] [Analyzer] [Error] Unhandled exception:
NoSuchMethodError: The getter 'iterator' was called on null.
Receiver: null
Tried calling: iterator
#0      Driver._captureExceptions.errorFunction (package:analysis_server/src/server/driver.dart:677:7)
#1      _CustomZone.handleUncaughtError (dart:async/zone.dart:1003:19)
#2      _CustomZone.runGuarded (dart:async/zone.dart:925:7)
#3      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#4      _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#5      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#6      _Timer._runTimers (dart:isolate-patch/timer_impl.dart:391:30)
#7      _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
#8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

**Comment by DanTup** [comment]
@scheglov this looks like the server crashing on startup - have you seen it before?

@mezoni can you provide the version of the SDK you're using? (output of running `/home/andrew/tools/dart-sdk/bin/dart --version`).

**Comment by mezoni** [comment]
After adding some files (about 5-6 files in just a few minutes), an endless stream of these messages began (The Dart Analyzer has terminated).
That is, any interactive action of the user, that requiring a call to the Dart Analyzer, caused this error message.
After moving these files outside the project, the analyzer started working normally.
Some of the files have already been returned to their place, the rest of the files are still outside the project.
It looks very strange.
Possible reasons.
- File structure
- File names
- Source code

Currently, there is no time to conduct an in-depth analysis of the cause. I am not a fan of such studies.

**Comment by mezoni** [comment]
Revert an extension and Dart SDK to previous versions does not solves the problem. This was a first attempt to solve the problem.
The problem is in the input data used by the analyzer.  Judging by causation

**Comment by scheglov** [comment]
Probably duplicate for https://github.com/dart-lang/sdk/issues/38261.

**Comment by DanTup** [comment]
Ah yes, it does look the same. I'll close this as a dupe of that. @mezoni the fix for the above has been shipping in v2.6.0 dev builds, so you could try one of those from https://dart.dev/tools/sdk/archive to confirm it solves your issue.
